packages/compass-saved-aggregations-queries/src/components/aggregations-queries-list.tsx

```diff
@@ -67,29 +69,34 @@ const AggregationsQueriesList = ({
     void fetchItems();
   }, [fetchItems]);
 
+  const [gridHeader, listItems] = useGridHeader(items);
+
+  const renderItem: React.ComponentProps<typeof VirtualGrid>['renderItem'] =
+    React.useCallback(
+      ({ index, ...props }: { index: number }) => (
+        <ConnectedItemCard index={index} items={listItems} {...props} />
```

Comment:
This component doesn't need to be aware of all the items in the list to be rendered. It's an anti-pattern in react to pass values like that through props if it can be avoided: it leads to unneccessary re-renders because unrelated values will change and blurs the borders between what is actually rendered in the view and what is not

If we can't do this through `connect`, we should rather get the props we need here and pass them to the component instead of passing all the items